docs: clarify timeout hierarchy in traffic shaping documentation (#8203)
apollographql/router: docs/source/routing/performance/traffic-shaping.mdx

If the router-level timeout is hit, the router returns a [HTTP 504 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/504) to indicate a gateway timeout.

#### Subgraph timeouts

Configure timeout values for requests between the router and subgraphs:

```yaml title="router.yaml"
traffic_shaping:
  all:
    timeout: 50s # If subgraph requests take more than 50 seconds, cancel the request (30 seconds by default)
```

Subgraph timeouts apply to each individual subgraph fetch request. There can be multiple subgraph requests, even to the same subgraph, within a single router request.

If the subgraph-level timeout is hit, the router returns a [GraphQL-compliant HTTP 200 status code](https://graphql.github.io/graphql-over-http/). Because the router can handle multiple subgraph requests for one operation, the timeout of one subgraph doesn't immediately invalidate other requests in the query plan. Returning a 200 status code enables the router to return partial data from other requests that didn't timeout.

#### Timeout hierarchy

> Set your router timeout to be greater than your subgraph timeouts. This enables the router to properly handle subgraph timeouts and return appropriate error responses to clients. Configure subgraph timeouts to be lower than the router timeout to provide processing buffer time.

Author: ```suggestion
Set your router timeout to be greater than your subgraph timeouts. This difference provides a processing buffer, which enables the router to properly handle subgraph timeouts and return appropriate error responses to clients.
```

    timeout: 50s # If subgraph requests take more than 50 seconds, cancel the request (30 seconds by default)
```

> When configuring timeouts, you should set your router timeout to be greater than or equal to your subgraph timeouts. This allows the router to properly handle subgraph timeouts and return appropriate error responses to clients. Ideally, subgraph timeouts should be slightly lower than the router timeout to provide a buffer for the router to process the timeout response.

Author: ```suggestion
When configuring timeouts, set your router timeout to be greater than or equal to your subgraph timeouts. This allows the router to properly handle subgraph timeouts and return appropriate error responses to clients. Ideally, subgraph timeouts should be slightly lower than the router timeout to provide a buffer for the router to process the timeout response.
```
